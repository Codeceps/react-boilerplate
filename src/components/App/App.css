@font-face {
    font-family: 'Pacifico';
    src: url('~resources/fonts/Pacifico.ttf');
}

#logo {
    background-color: black;
}

#logo-img {
    widows: 120px;
    height: 120px;
    margin: auto;
    display: block;
}

.message {
    font-family: 'Pacifico';
    color: cadetblue;
    text-align: center;
}

.count {
    font-family: 'Pacifico';
    font-size: 120%;
    color: chocolate;
    text-align: center;
}

.button {
    margin: auto;
    display: block;
}