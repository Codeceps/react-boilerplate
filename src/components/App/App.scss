@import '~resources/css/Variables';

@font-face {
    font-family: 'Pacifico';
    src: url('~resources/fonts/Pacifico.ttf');
}

@media only screen and (max-width: 600px) {
    body {
        background-color: black;
    }
}


:root {
    --mainColor: #000000ff;
}

#logo-container {
    background-color: var(--mainColor);
}

#logo-img {
    width: $size;
    height: $size;
    margin: auto;
    display: block;
}

#header-text {
    font-family: 'Pacifico';
    font-size: 2em;
    text-align: center;
    color: #61dafbff;
    background-color: black;
}

@custom-selector :--cw .centered-widget;

:--cw {
    margin-top: 10px;
    margin-bottom: 10px;
    margin-left: auto;
    margin-right: auto;
    display: block;
}

::placeholder {
    color: gray;
}