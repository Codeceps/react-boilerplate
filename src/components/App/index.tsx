import React from 'react';
import {hot} from 'react-hot-loader/root';
//@ts-ignore
import react_logo_svg from 'resources/images/react_logo.svg';


import './App.css';


export interface AppProps {
    message: string
}

export interface AppState {
    clickCount: number
}

class App extends React.Component<AppProps, AppState> {

    state: AppState = {
        clickCount: 0
    }

    constructor(props: AppProps) {
        super(props);
    }


    clickListener = () => {
        this.setState((prev) => ({
            clickCount: prev.clickCount + 1
        }))
    }

    render() {
        const {clickCount} = this.state;
        
        return (
            <div>
                <div id='logo'>
                    <img id='logo-img' src={react_logo_svg}/>
                </div>

                <div id="content">
                    <h1 className='message'>{this.props.message}</h1>

                    <p className="count">Click count: {clickCount}</p>
                    <button className='button' onClick={this.clickListener}>Click Me</button>
                </div>   
            </div>
        )
    }

}


export default hot(App);