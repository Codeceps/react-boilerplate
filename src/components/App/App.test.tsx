import React from "react";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './';


Enzyme.configure({adapter: new Adapter()});

describe('<App />', () => {

    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App message=''/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    /**
     * Snapshot test using 'react-test-renderer'
     */
    it('should have valid snapshot', () => {
        const component = renderer.create(<App message='' />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    /**
     * DOM test using 'enzyme'
     */
    test('jest mocked the react logo svg image used', () => {
        const component = shallow(<App message='' />);

        const imageSource = component.find('img').prop('src');
        expect(imageSource).toBe('test-file-stub');
    });

})