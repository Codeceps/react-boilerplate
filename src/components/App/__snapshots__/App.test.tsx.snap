// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should have valid snapshot 1`] = `
<div>
  <div
    id="logo-container"
  >
    <img
      id="logo-img"
      src="test-file-stub"
    />
  </div>
  <div
    id="header-text"
  >
    
  </div>
  <div>
    <ul
      id="nav"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/lazy-test"
          onClick={[Function]}
        >
          Lazy Test
        </a>
      </li>
    </ul>
  </div>
  <div>
    <h2
      id="home-title"
    >
      Stored text: 
      No Text Stored
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        className="centered-widget"
        onChange={[Function]}
        placeholder="Enter text"
        type="text"
        value=""
      />
      <input
        className="centered-widget"
        type="submit"
        value="Store text"
      />
    </form>
  </div>
</div>
`;
